packages/headless/src/controllers/result-list/headless-result-link-analytics.ts

```diff
@@ -0,0 +1,77 @@
+import {Result} from '../../api/search/search/result';
+import {Engine} from '../../app/headless-engine';
+import {logDocumentOpen} from '../../features/result/result-analytics-actions';
+
+// 1 second is a reasonable amount of time to catch most longpress actions
+const defaultDelay = 1000;
+
+type ResultLinkAnalyticsOptions = {
+  /**
+   * The result for which to log analytics.
+   */
+  result: Result;
+  /**
+   * The amount of time to wait before selecting the result after calling `beginDelayedSelect`.
+   */
+  delay?: number;
+};
+
+type ResultLinkAnalyticsProps = {
+  /** The options for the `ResultLinkAnalytics` controller. */
+  options: ResultLinkAnalyticsOptions;
+};
+
+/**
+ * The `ResultLinkAnalytics` headless controller helps more easily send analytics when a search result is selected.
```

Comment:
Suggestion:

> The `ResultLink` controller provides an interface for triggering desirable side effects, such as logging UA events to the Coveo Platform, when a user selects a query result.

Reply:
Or if we're sure that the controller will only be used to ensure that UA events are logged:

> The `ResultLink` controller provides an interface for automatically logging UA events to the Coveo Platform as a side effect when a user selects a query result.